code/modules/antagonists/blob/blobstrains/reactive_spines.dm

```diff
@@ -17,8 +17,15 @@
 		COOLDOWN_START(src, retaliate_cooldown, 2.5 SECONDS) // 2.5 seconds before auto-retaliate can whack everything within 1 tile again
 		B.visible_message(span_boldwarning("The blob retaliates, lashing out!"))
 		for(var/atom/A in range(1, B))
+			if(HAS_TRAIT(A, TRAIT_MAGICALLY_PHASED))
+				continue
 			var/attacked_turf = get_turf(A)
-			if(isliving(A) && !isblobmonster(A)) // Make sure to inject strain-reagents with automatic attacks when needed.
+			var/mob/living/living_mob
+			if(isliving(A))
+				living_mob = A
+				if(living_mob.incorporeal_move)
+					continue
+			if(living_mob && !isblobmonster(A)) // Make sure to inject strain-reagents with automatic attacks when needed.
```

Comment:
Could you put these checks in a seperate proc? They're copypasted throughout blobcode now

Reply:
Not sure what you mean, each one of these has separate conditions

Reply:
![image](https://user-images.githubusercontent.com/6209658/206585881-b2227319-caaf-4954-9e13-dd00076bc7bf.png)
do you want this

cause i hate this, but i'll do it if you really want.